components/places/src/bookmark_sync/engine.rs

```diff
@@ -373,6 +385,35 @@ fn update_local_items_in_places(
     scope.err_if_interrupted()?;
     db.execute_batch("DELETE FROM applyNewLocalStructureOps")?;
 
+    // Similar to the check in apply_remote_items, however we do a post check
+    // to see if dogear was unable to fix up the issue
+    let orphaned_count: i64 = db.query_row(
+        "WITH RECURSIVE orphans(id) AS (
+           SELECT b.id
+           FROM moz_bookmarks b
+           WHERE b.parent IS NOT NULL
+             AND NOT EXISTS (
+               SELECT 1 FROM moz_bookmarks p WHERE p.id = b.parent
+             )
+           UNION
+           SELECT c.id
+           FROM moz_bookmarks c
+           JOIN orphans o ON c.parent = o.id
+         )
+         SELECT COUNT(*) FROM orphans;",
+        [],
+        |row| row.get(0),
+    )?;
+
+    if orphaned_count > 0 {
```

Comment:
It might be tricky to use sentry to tie together how many users see one or both of these, but that's fine for now.